misc-utils/kill.c

```diff
@@ -286,24 +354,58 @@ static char **parse_arguments(int argc, char **argv, struct kill_control *ctl)
 				errx(EXIT_FAILURE, _("too many arguments"));
 			/* argc == 2, accept "kill -l $?" */
 			arg = argv[1];
+			if (arg[0] == '0' && arg[1] == 'x') {
+				uint64_t sigmask;
+				if (ul_strtou64(arg + 2, &sigmask, 16) < 0)
+					errx(EXIT_FAILURE, _("wrong sigmask format: %s"), arg);
+				print_signal_mask(sigmask, '\n');
+				exit(EXIT_SUCCESS);
+			}
 			if ((ctl->numsig = arg_to_signum(arg, 1)) < 0)
 				errx(EXIT_FAILURE, _("unknown signal: %s"),
 				     arg);
-			print_signal_name(ctl->numsig);
+			print_signal_name(ctl->numsig, true);
 			exit(EXIT_SUCCESS);
 		}
 		/* for compatibility with procps kill(1) */
 		if (!strncmp(arg, "--list=", 7) || !strncmp(arg, "-l=", 3)) {
 			char *p = strchr(arg, '=') + 1;
+			if (p[0] == '0' && p[1] == 'x') {
+				uint64_t sigmask;
+				if (ul_strtou64(p + 2, &sigmask, 16) < 0)
+					errx(EXIT_FAILURE, _("wrong sigmask format: %s"), p);
+				print_signal_mask(sigmask, '\n');
+				exit(EXIT_SUCCESS);
+			}
 			if ((ctl->numsig = arg_to_signum(p, 1)) < 0)
 				errx(EXIT_FAILURE, _("unknown signal: %s"), p);
-			print_signal_name(ctl->numsig);
+			print_signal_name(ctl->numsig, true);
 			exit(EXIT_SUCCESS);
 		}
 		if (!strcmp(arg, "-L") || !strcmp(arg, "--table")) {
 			print_all_signals(stdout, 1);
 			exit(EXIT_SUCCESS);
 		}
+		if (!strcmp(arg, "-d") || !strcmp(arg, "--show-process-state")) {
+			uint64_t pid;
+			if (argc < 2)
+				errx(EXIT_FAILURE, _("too few arguments"));
+			if (2 < argc)
+				errx(EXIT_FAILURE, _("too many arguments"));
+			arg = argv[1];
+			if (ul_strtou64(arg, &pid, 10) < 0)
+				errx(EXIT_FAILURE, _("wrong pid specification: %s"), arg);
```

Comment:
pid = strtou64_or_err(argv[1], _("invalid pid argument"));

Reply:
With the change to `pid_t`: `strtopid_or_err()`

Reply:
I didn't know strtopid_or_err. Thank you.

Reply:
Fixed in locally.